src/DynamoCoreWpf/Utilities/NodeContextMenuBuilder.cs

```diff
@@ -0,0 +1,451 @@
+                command: new Binding
+                {
+                    Source = NodeViewModel,
+                    Path = new PropertyPath("SetLacingTypeCommand"),
+                },
+                commandParameter: "CrossProduct",
+                isCheckable: true,
+                isChecked: new Binding
+                {
+                    Source = NodeViewModel,
+                    Path = new PropertyPath("ArgumentLacing"),
+                    Converter = new EnumToBooleanConverter(),
+                    ConverterParameter = "CrossProduct",
+                    Mode = BindingMode.OneWay
+                }
+            ));
+
+            return lacingMenuItem;
+        }
+
+        // To be connected in a future PR for Node Info States.
+        private static MenuItem BuildDismissedAlertsMenuItem()
+        {
+            //MenuItem dismissedAlertsMenuItem = CreateMenuItem
+            //(
+            //    name: "dismissedAlerts",
+            //    header: Wpf.Properties.Resources.NodeInformationalStateDismissedAlerts,
+            //    itemsSource: new Binding
+            //    {
+            //        Source = viewModel,
+            //        Path = new PropertyPath("DismissedAlerts"),
+            //        UpdateSourceTrigger = UpdateSourceTrigger.PropertyChanged
+            //    }
+            //);
+            //DataTemplate itemTemplate = new DataTemplate(typeof(MenuItem));
+
+            //FrameworkElementFactory grid = new FrameworkElementFactory(typeof(Grid));
+            //grid.Name = "mainTemplateGrid";
+            //grid.SetValue(WidthProperty, 220.0);
+            //grid.SetValue(HeightProperty, 30.0);
+            //grid.SetValue(MarginProperty, new Thickness(-15,0,0,0));
+            //grid.SetValue(HorizontalAlignmentProperty, HorizontalAlignment.Left);
+            //grid.SetValue(VerticalAlignmentProperty, VerticalAlignment.Stretch);
+            //grid.SetValue(BackgroundProperty, new SolidColorBrush(Colors.Transparent));
+
+            //FrameworkElementFactory textBlock = new FrameworkElementFactory(typeof(TextBlock));
+            //textBlock.SetValue(HorizontalAlignmentProperty, HorizontalAlignment.Stretch);
+            //textBlock.SetValue(MarginProperty, new Thickness(15, 0, 0, 0));
+            //textBlock.SetValue(IsHitTestVisibleProperty, false);
+            //textBlock.SetValue(VerticalAlignmentProperty, VerticalAlignment.Center);
+            //textBlock.SetValue(ForegroundProperty, new SolidColorBrush(Color.FromRgb(238,238,238)));
+
+            //dismissedAlertsMenuItem.ItemTemplate = itemTemplate;
+
+            //AddContextMenuItem(dismissedAlertsMenuItem, insertionPoint++);
+
+            return null;
+        }
+
+        private static void AddInjectedNodeViewCustomizationMenuItems(Dictionary<string, object> nodeViewCustomizationMenuItems)
+        {
+            foreach (KeyValuePair<string, object> keyValuePair in nodeViewCustomizationMenuItems)
```

Comment:
var

Reply:
var doesn't let me grab the Key or Value, which is needed here. 
I've swapped for:

`foreach (DictionaryEntry keyValuePair in nodeViewCustomizationMenuItems)
            {
                ContextMenu.Items.Add(keyValuePair.Value);
            }`

Reply:
`var` doesn't change the type, the type is already inferred from the type of `nodeViewCustomizationMenuItems` - it's a KeyValuePair no matter what.

Reply:
I don't think it will let me access the Value property if it's 'var' though:

![image](https://user-images.githubusercontent.com/29973601/135314263-493cddb3-5144-4fde-aeef-8f4adba48faa.png)

Reply:
try compiling that - I'm guessing resharper is just getting confused- for example:
https://github.com/DynamoDS/Dynamo/blob/b17e9843a6ce0c3568c4a20fb960ff631cf103e8/test/DynamoCoreTests/SerializationTests.cs#L238


and an example:
https://dotnetfiddle.net/nekXuD